course_discovery/apps/course_metadata/signals.py

```diff
@@ -325,3 +325,45 @@ def check_course_runs_within_course_for_duplicate_external_key(course, specific_
         if external_key == specific_course_run.external_key and course_run != specific_course_run:
             message = _duplicate_external_key_message([course_run])
             raise ValidationError(message)
+
+
+# This is code to handle signals emitted from the event bus. This is currently test code and not to be used
+# for production use cases.
+try:  # pragma: no cover
+    from edx_toggles.toggles import SettingToggle  # pylint: disable=import-outside-toplevel
+    from openedx_events.content_authoring.data import CourseCatalogData  # pylint: disable=import-outside-toplevel
+    from openedx_events.content_authoring.signals import \
+        COURSE_CATALOG_INFO_CHANGED  # pylint: disable=import-outside-toplevel
+
+    # .. toggle_name: LOG_KAFKA_EVENTS
+    # .. toggle_implementation: SettingToggle
+    # .. toggle_default: False
+    # .. toggle_description: Enables the logging of events from the Kafka event bus
+    # .. toggle_use_cases: opt_in
+    # .. toggle_creation_date: 2022-08-11
+    # .. toggle_tickets: https://github.com/edx/edx-arch-experiments/issues/48
+    LOG_KAFKA_EVENTS = SettingToggle('LOG_KAFKA_EVENTS', default=False)
+
+    @receiver(COURSE_CATALOG_INFO_CHANGED)
+    def log_event_from_event_bus(**kwargs):
```

Comment:
Would there be any problem with doing this instead?

```suggestion
    def log_event_from_event_bus(*, catalog_info, **kwargs):
```

This would allow other kwargs but always extract the `catalog_info` param (and require it to be present, which should be a pretty safe bet).

Reply:
I'm OK with having just `**kwargs` if all of this is expected to be subject to change, just curious.

Reply:
Yeah, this was just the pattern from the previous logging code. This code is going to go away as soon as we confirm we're getting event data correctly.

Reply:
Cool. I'm fine with taking it cautious at first.